main.rb

```diff
@@ -0,0 +1,122 @@
+module Enumerable
+  def my_each
+    return unless block_given?
+
+    each do |i|
+      yield i
+    end
+    self
+  end
+
+  def my_each_with_index
+    return unless block_given?
+
+    0.upto(length - 1) do |i|
+      yield i, index
+      index + 1
+    end
+  end
+
+  def my_select
+    return unless block_given?
+
+    result = []
+    my_each do |i|
+      result.push(i) if yield i
+    end
+    result
+  end
```

Comment:
# my_select

- [ ] `my_select` returns an enumerator if no block is given

     Failure/Error: expect(array.my_select).to be_an(Enumerator)
       expected nil to be a kind of Enumerator